pkg/pod/wait.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pod
+
+import (
+	"time"
+
+	"golang.org/x/net/context"
+)
+
+// WaitExited waits for a pod to (run and) exit.
+// This method blocks indefinitely and refreshes the pod state.
+func (p *Pod) WaitExited() error {
+	// isExited implies isExitedGarbage.
+	for !p.isExited && !p.isAbortedPrepare && !p.isGarbage && !p.isGone {
```

Comment:
similar to `p.AfterRun`, a helper of `p.TerminalState` could make this clearer..

Also, technically this isn't technically a "wait exited" because `isAbortedPrepare` is a pre-run, but terminal, state, so this is really `WaitFinished` or `WaitTerminal`